Spigot-Server-Patches/0581-Add-Village-API.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TheShermanTanker <tanksherman27@gmail.com>
+Date: Thu, 24 Sep 2020 19:01:07 +0800
+Subject: [PATCH] Add-Village-API
+
+
+diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
+index 95da2a56028e448c3a41115501f25a20382961f7..94ae395bd29e11bf7c90d6fe503b23fa3e78820a 100644
+--- a/src/main/java/net/minecraft/server/WorldServer.java
++++ b/src/main/java/net/minecraft/server/WorldServer.java
+@@ -1962,10 +1962,18 @@ public class WorldServer extends World implements GeneratorAccessSeed {
+         return this.getChunkProvider().j();
+     }
+ 
++    public VillagePlace getVillageManager() { //Paper - OBFHELPER
++    	return this.y();
++    }
++
+     public boolean a_(BlockPosition blockposition) {
+         return this.a(blockposition, 1);
+     }
+ 
++    public boolean isVillage(BlockPosition blockpos) { //Paper - OBFHELPER
++    	return this.a_(blockpos);
++    }
++
+     public boolean a(SectionPosition sectionposition) {
+         return this.a_(sectionposition.q());
+     }
+@@ -1974,10 +1982,18 @@ public class WorldServer extends World implements GeneratorAccessSeed {
+         return i > 6 ? false : this.b(SectionPosition.a(blockposition)) <= i;
+     }
+ 
++    public boolean isCloseToVillage(BlockPosition blockpos, int distanceInChunks) { //Paper - OBFHELPER
++    	return this.a(blockpos, distanceInChunks);
++    }
++
+     public int b(SectionPosition sectionposition) {
+         return this.y().a(sectionposition);
+     }
+ 
++    public int sectionsToVillage(SectionPosition sectionpos) { //Paper - OBFHELPER
++    	return this.b(sectionpos);
++    }
++
+     public PersistentRaid getPersistentRaid() {
+         return this.persistentRaid;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 299f57ca2a65887a0d7e7c584fc1bd5c783b0db2..54f57ff537f25e3d99f11285cabfd5445fa0dec0 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -86,6 +86,7 @@ import net.minecraft.server.PlayerChunk;
+ import net.minecraft.server.ProtoChunkExtension;
+ import net.minecraft.server.RayTrace;
+ import net.minecraft.server.SavedFile;
++import net.minecraft.server.SectionPosition;
+ import net.minecraft.server.SoundCategory;
+ import net.minecraft.server.StructureGenerator;
+ import net.minecraft.server.Ticket;
+@@ -2609,4 +2610,20 @@ public class CraftWorld implements World {
+         return spigot;
+     }
+     // Spigot end
++
++	@Override
++	public boolean isVillage(int x, int y, int z) {
++		return world.isVillage(new BlockPosition(x, y, z));
```

Comment:
_cries in tabs_

Reply:
At this point that's my IDE's fault since I left all the proper spacing in there

The patch system really hates me I guess :/

Reply:
If you're using IJ make sure you have it set to adapt to project codestyle.

If you're not using IntelliJ, start using IntelliJ